cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -297,18 +298,33 @@ func (c *client) CreateMachineSetObjects(machineSets []*clusterv1.MachineSet, na
 }
 
 func (c *client) CreateMachineObjects(machines []*clusterv1.Machine, namespace string) error {
+	var (
+		wg      sync.WaitGroup
+		errOnce sync.Once
+		gerr    error
```

Comment:
Is the error returned from `Wait()` not useful? If there are multiple errors from this routine, this will only print one of them. Looking at the source where you linked I see that it does the same, which is unfortunate. It'd be nicer if the errors were coalesced.

Reply:
Is there some standard way of coalescing errors in the project?

Reply:
No. I've written one before, so I know it's possible, but we don't have any such library at the moment. Probably not worth doing right now. I didn't realize that the groups thing also threw away everything but the first error when I first saw this, so this is more of a comment about the disappointing state of error handling than something actionable for you in this PR.

Reply:
Gotcha, I could use https://github.com/hashicorp/go-multierror but it probably isn't worth a new dependency.